plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -557,10 +557,8 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
     public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
-        Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
-
         ImmutableMap.Builder<String, ColumnHandle> columnHandles = ImmutableMap.builder();
-        for (IcebergColumnHandle columnHandle : getColumns(icebergTable.schema(), typeManager)) {
+        for (IcebergColumnHandle columnHandle : getColumns(SchemaParser.fromJson(table.getTableSchemaJson()), typeManager)) {
```

Comment:
this is a good change.

However, it looks like we call `SchemaParser.fromJson(tableHandle.getTableSchemaJson())` multiple times on one table handle. Am i right?

`SchemaParser.fromJson` does cache internally (on a static field).
This isn't ideal, and we could better, caching within table handle object. Not sure it matters though -- depends how frequently this is called.

Reply:
Should we swtich to `SchemaParser.fromJson(JsonNode)` 


```
fromJson(JsonUtil.mapper().readValue(jsonKey, JsonNode.class))
```

?